### PR TITLE
Hwp 329 - create file for testing API community requests

### PR DESCRIPTION
**Pull request by mattsiel**
# Description

This PR includes the following proposed change(s):

- Ticket number 329
- created a file with all requests included in the communities section of swagger
- updated get Communities to use the requests from the new community requests file
- created a folder for API requests to be used in tests
- updated jest config to be able to read the environment variables

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [ ] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [ ] Tested by Zach.
- [ ] Tested by Brandon.
- [ ] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

File: app/server/tests/api-getCommunity.test.js [hunk 1]
```diff
@@ -1,135 +1,91 @@
...
 
-  test('API returns description "Forbidden."', async () => {
-    let response = await request
-      .get('/community')
-      .set({authorization: `Bearer ${token}11`})
-
+  test('API returns description -  "Forbidden."', () => {
     expect(response.text).toBe("Forbidden.");
   });
 });
```
Comment (mattsiel) [inline app/server/tests/api-getCommunity.test.js:91]:
```suggestion
});

```

File: app/server/tests/functions/communityFunctions.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+function getRulesForCommunity(title,token){
+    return request
+        .get(`/community/rules/${title}`)
+        .set({authorization: `Bearer ${token}`})
+}
+
+module.exports = {getCommunities, getCommunitybyTitle,createCommunity,deleteCommunity,
+    patchCommunitybyTitle,joinCommunitybyTitle,leaveCommunitybyTitle,setRulesForCommunity,
+    getRulesForCommunity };
```
Comment (mattsiel) [inline app/server/tests/functions/communityFunctions.js:65]:
```suggestion
    getRulesForCommunity };
    
```

File: app/server/tests/functions/communityFunctions.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+require('dotenv').config();
+const supertest = require('supertest');
+const endpoint = process.env.API_REF;
+const request = supertest(endpoint);
+
+function getCommunities(token){
```
Comment (BradyMitch) [inline app/server/tests/functions/communityFunctions.js:6]: I really like how simple these functions are. Easy to understand.

Comment (mattsiel) [reply app/server/tests/functions/communityFunctions.js:6]: thanks, we will do this with all the requests that are documented on swagger from now on

**Review by dbarkowsky — APPROVED** [review]

**Pull request merged**